File: samples/snippets/visualbasic/VS_Snippets_CLR/CodeConditionStatementExample/VB/codeconditionstatementexample.vb
 '<Snippet1>
Imports System
Imports System.CodeDom

Namespace CodeDomSamples
    
   Public Class CodeConditionStatementExample
      
      Public Sub New()
          '<Snippet2>
          ' Create a CodeConditionStatement that tests a boolean value named boolean.
           Dim conditionalStatement As New CodeConditionStatement( _
                New CodeVariableReferenceExpression("boolean"), _
                New CodeStatement() {New CodeCommentStatement("If condition is true, execute these statements.")}, _
                New CodeStatement() {New CodeCommentStatement("Else block. If condition is false, execute these statements.")})

          ' A Visual Basic code generator produces the following source code for the preceeding example code:

          ' If [boolean] Then
          '     'If condition is true, execute these statements.
          ' Else
          '     'Else block. If condition is false, execute these statements.
          
          '</Snippet2>
      End Sub 'New 

   End Class 'CodeConditionStatementExample

End Namespace 'CodeDomSamples 		
        
'</Snippet1>

File: samples/snippets/csharp/VS_Snippets_CLR/CodeConditionStatementExample/CS/codeconditionstatementexample.cs
//<Snippet1>
using System;
using System.CodeDom;

namespace CodeDomSamples
{
    public class CodeConditionStatementExample
    {
        public CodeConditionStatementExample()
        {
            //<Snippet2>
            // Create a CodeConditionStatement that tests a boolean value named boolean.
            CodeConditionStatement conditionalStatement = new CodeConditionStatement(
                // The condition to test.
                new CodeVariableReferenceExpression("boolean"),
                // The statements to execute if the condition evaluates to true.
                new CodeStatement[] { new CodeCommentStatement("If condition is true, execute these statements.") },
                // The statements to execute if the condition evalues to false.
                new CodeStatement[] { new CodeCommentStatement("Else block. If condition is false, execute these statements.") } );

            // A C# code generator produces the following source code for the preceeding example code:

            // if (boolean) 
            // {
                //     // If condition is true, execute these statements.
            // }
            // else {
            //     // Else block. If condition is false, execute these statements.
                // }        
    
            //</Snippet2>
        }
    }
}
//</Snippet1>